packages/hub/tasks/scheduled-payment-on-chain-cancelation-waiter.ts

```diff
@@ -0,0 +1,99 @@
+    let scheduledPayment = await prisma.scheduledPayment.findFirstOrThrow({
+      where: { id: payload.scheduledPaymentId },
+    });
+
+    let provider = this.ethersProvider.getInstance(scheduledPayment.chainId);
+    let scheduledPaymentModule = await this.cardpay.getSDK('ScheduledPaymentModule', provider);
+
+    if (scheduledPayment.cancelationBlockNumber) {
+      // Did we somehow spawn this task after it already updated cancelationBlockNumber previously?
+      return Sentry.captureException(
+        new Error('Task run for scheduled payment that already has a cancelation block number'),
+        {
+          tags: {
+            action: 'scheduled-payment-task',
+            scheduledPaymentId: scheduledPayment.id,
+          },
+        }
+      );
+    }
+
+    if (!scheduledPayment.cancelationTransactionHash) {
+      // This should never happen because we spawn this task only after updating cancelationTransactionHash in the route. But just in case...
+      return Sentry.captureException(new Error('Missing transaction hash'), {
+        tags: {
+          action: 'scheduled-payment-task',
+          scheduledPaymentId: scheduledPayment.id,
+        },
+      });
+    }
+
+    if (isBefore(scheduledPayment.canceledAt!, subDays(nowUtc(), 1))) {
+      return Sentry.captureException(
+        new Error(
+          'Scheduled payment has been canceled more than a day ago and the cancelation transaction still has not been mined'
+        ),
+        {
+          tags: {
+            action: 'scheduled-payment-task',
+            scheduledPaymentId: scheduledPayment.id,
+          },
+        }
+      );
+    }
+
+    try {
+      let receipt = await scheduledPaymentModule.cancelPaymentOnChain(scheduledPayment.cancelationTransactionHash);
+      return await prisma.scheduledPayment.update({
+        data: {
+          cancelationBlockNumber: receipt.blockNumber, // To let us know that the scheduled payment has been created on chain and that the process of scheduling the payment was completed
+        },
+        where: {
+          id: scheduledPayment.id,
+        },
+      });
+    } catch (error: any) {
+      if (error.message.includes('revert')) {
+        // Error message from the waitUntilTransactionMined in the SDK will be: Transaction with hash "${txnHash}" was reverted
+        await prisma.scheduledPayment.update({
+          data: {
+            cancelationTransactionError: error.message,
+          },
+          where: {
+            id: scheduledPayment.id,
+          },
+        });
+
+        return; // We don't want to throw an error here because we don't want the task to be retried - the transaction was obviously reverted
+      }
+
+      // At this point, we don't know what error to expect. Let's throw the error so that the task will restart
+      Sentry.captureException(error, {
+        tags: {
+          action: 'scheduled-payment-task',
+          scheduledPaymentId: scheduledPayment.id,
+        },
+      });
```

Comment:
You can make assertions to ensure things are captured to Sentry [like this](https://github.com/cardstack/cardstack/blob/a9e9b09cd2ec0e6ef5de354aae64536335b7e0ea/packages/hub/node-tests/tasks/subscribe-email-test.ts#L83-L90) if you like

Reply:
Nice!